superchain/superchain.go

```diff
@@ -99,10 +99,52 @@ type ChainConfig struct {
 	// This matches the resource filename, it is not encoded in the config file itself.
 	Chain string `yaml:"-"`
 
+	// UsePlasma is activated when the chain is in plasma mode.
+	UsePlasma    *bool        `json:"use_plasma,omitempty" yaml:"use_plasma,omitempty"`
+	Plasma       PlasmaConfig `json:"plasma,omitempty" yaml:"plasma,omitempty"`
+	PlasmaConfig `yaml:",inline"`
+
 	// Hardfork Configuration Overrides
 	HardForkConfiguration `yaml:",inline"`
 }
 
+type PlasmaConfig struct {
+	DAChallengeAddress *Address `json:"da_challenge_contract_address,omitempty" yaml:"da_challenge_contract_address,omitempty"`
+	// DA challenge window value set on the DAC contract. Used in plasma mode
+	// to compute when a commitment can no longer be challenged.
+	DAChallengeWindow *uint64 `json:"da_challenge_window,omitempty" yaml:"da_challenge_window,omitempty"`
+	// DA resolve window value set on the DAC contract. Used in plasma mode
+	// to compute when a challenge expires and trigger a reorg if needed.
+	DAResolveWindow *uint64 `json:"da_resolve_window,omitempty" yaml:"da_resolve_window,omitempty"`
+}
+
+func (c *ChainConfig) CheckPlasma() error {
```

Comment:
Is it a bit surprising that a function called `CheckFoo` actually ends up mutating the receiver? See other comment.

Reply:
Would it suffice to just rename this function to better describe its functionality? Perhaps `PlasmaCheckAndTransform`? If not, would you prefer to have separate structs for `Unmarshaling` from the `rollup.json` versus writing to into the .yaml file?

Reply:
I think renaming would be good. Don't have strong opinions on separate structs, but just keen to leave all data structures in a neat state which won't cause confusion downstream.

Reply:
Good call. Ended up using a separate struct for `Unmarshal` to reduce downstream confusion risk: https://github.com/ethereum-optimism/superchain-registry/pull/224/commits/c038f4c9529edb19c328c51ace862ec818d4ae3c